packages/xstate-wallet/src/chain.ts

```diff
@@ -321,6 +321,9 @@ export class ChainWatcher implements Chain {
   }
 
   public get selectedAddress(): string | null {
+    if (this.mySelectedAddress === null && window && window.ethereum) {
+      return window.ethereum.selectedAddress ?? null;
+    }
```

Comment:
This *might* be a small improvement (if this class has no selectedAddress, try to fall back on `window.ethereum.selectedAddress`), but 

i) I don't think it will help solve the problem in the description of this PR (which is when `window.ethereum.selectedAddress` is not yet defined. 
ii) the change has little to do with the title of this PR. 

Also, I am a bit concerned that this introduces a path for the following potentially confusing situation: the private variable `this.mySelectedAddress` is null but `chain.selectedAddress` is a string. 

It might be a bit nicer to put this fallback logic inside the `ethereumIsEnabled()` getter below.

Reply:
I think it does fix the issue in the PR, because the value is only undefined for a split second right when the wallet loads

Reply:
I've updated it to _set_ `this. mySelectedAddress`.